### PR TITLE
BREAKING CHANGE: Remove Tag deprecation #DS-648

### PR DESCRIPTION
**Pull request by crishpeen**
## Description

Remove `light` and `default` modifiers in the `web` package. Use `subtle` and `neutral` instead.

Remove the `theme` prop in the `twig` package, use `isSubtle` instead. Remove color `default`, use `neutral instead.

Remove the `theme` prop in the `react` package, use `isSubtle` instead. Remove color `default`, use `neutral instead. Remove the `tag` prop, use `elementType` instead.

### Additional context

All removed props were deprecated. See each commit message for the migration guide.

### Issue reference

#DS-648

---

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-desing-system/blob/main/CONTRIBUTING.md).
- [x] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-desing-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/web-react/src/components/Tag/__tests__/Tag.test.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ describe('Tag', () => {
     expect(element.textContent).toBe('Tag');
   });
 
-  it.each([['neutral'], ['default']])('should render color %s', (color) => {
+  it.each([['neutral']])('should render color %s', (color) => {
```
Comment (literat) [inline packages/web-react/src/components/Tag/__tests__/Tag.test.tsx:35]: I think that the  `each` is not necessary anymore in this case.

File: packages/web-react/src/components/Tag/__tests__/useTagStyleProps.test.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ describe('useTagStyleProps', () => {
     expect(result.current.classProps).toBe('Tag');
   });
 
-  it.each([['neutral'], ['default'], ['success'], ['informative'], ['warning'], ['danger']])(
+  it.each([['neutral'], ['success'], ['informative'], ['warning'], ['danger']])(
```
Comment (literat) [inline packages/web-react/src/components/Tag/__tests__/useTagStyleProps.test.ts:13]: Can we use one of the dictionary prop tests?

**Review by literat — COMMENTED** [review]
Nice work, but we can do better on some of the tests, I think.

File: packages/web-react/src/components/Tag/__tests__/useTagStyleProps.test.ts (hunk 2 above)
Comment (crishpeen) [reply packages/web-react/src/components/Tag/__tests__/useTagStyleProps.test.ts:13]: This is the hook test. AFAIK we currently don't have global tests for these. (We have them for the components tests)

Comment (literat) [reply packages/web-react/src/components/Tag/__tests__/useTagStyleProps.test.ts:13]: Ok, my fault.

**Review by tomassychra — APPROVED** [review]

**Review by literat — APPROVED** [review]
Nice, thank you 👍

File: packages/web-twig/src/Resources/components/Tag/README.md [hunk 3]
```diff
@@ -31,13 +31,12 @@ Without lexer:
...
-| `theme`       | `light`, `dark`                                                                 | `null`    | no       | [**DEPRECATED**][deprecated] in favor of `isSubtle`; Theme variant                                               |
+| Prop name     | Type                                                    | Default   | Required | Description                                |
+| ------------- | ------------------------------------------------------- | --------- | -------- | ------------------------------------------ |
+| `color`       | [Emotion Color dictionary][dictionary-color], `neutral` | `neutral` | no       | Color of the component                     |
+| `isSubtle`    | `boolean`                                               | `false`   | no       | Whether is Tag displayed in subtle variant |
+| `elementType` | `string`                                                | `span`    | no       | HTML tag to render                         |
+| `size`        | [Size Extended dictionary][dictionary-size]             | `medium`  | no       | Size of the Tag                            |
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/Tag/README.md:39]: 🔤 ?

File: packages/web-twig/src/Resources/components/Tag/README.md [hunk 4]
```diff
@@ -31,13 +31,12 @@ Without lexer:
...
-| `size`        | [Size Extended dictionary][dictionary-size]                                     | `medium`  | no       | Size of the Tag                                                                                                  |
-| `theme`       | `light`, `dark`                                                                 | `null`    | no       | [**DEPRECATED**][deprecated] in favor of `isSubtle`; Theme variant                                               |
+| Prop name     | Type                                                    | Default   | Required | Description                                |
+| ------------- | ------------------------------------------------------- | --------- | -------- | ------------------------------------------ |
+| `color`       | [Emotion Color dictionary][dictionary-color], `neutral` | `neutral` | no       | Color of the component                     |
+| `isSubtle`    | `boolean`                                               | `false`   | no       | Whether is Tag displayed in subtle variant |
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/Tag/README.md:37]: I guess this should be:

```suggestion
| `isSubtle`    | `boolean`                                               | `false`   | no       | Whether the Tag is displayed in subtle variant |
```

**Review by adamkudrna — APPROVED** [review]

**Pull request merged**